src/main/java/com/aws/greengrass/clientdevices/auth/CertificateManager.java

```diff
@@ -174,24 +169,23 @@ public void subscribeToCertificateUpdates(GetCertificateRequest getCertificateRe
                     getCertificateRequest.getCertificateRequestOptions().getCertificateType();
             // TODO: Should be configurable
             KeyPair keyPair = CertificateStore.newRSAKeyPair(4096);
-            X509Certificate[] caCertificates = getX509CACertificates();
 
             if (certificateType.equals(GetCertificateRequestOptions.CertificateType.SERVER)) {
-                Consumer<X509Certificate> consumer = (t) -> {
+                BiConsumer<X509Certificate, X509Certificate[]> consumer = (t, caCertificates) -> {
```

Comment:
Why a BiConsumer?

When a component subscribes we generate create a consumer that before, was referencing a variable outside of it, effectively creating a `closure`, even if the values returned by `getX509CACertificates` changed we were still holding on to a stale reference which is not good if the CA chains get changes by a configuration update.

Reply:
What's the difference between the 2 arguments of the consumer?

Also, this is an API breaking change, this impacts Moquette

Reply:
I know this CR is closed but to answer your questions:

1) t is the certificate got generated, caChain is the caChain (it only has CA certificates not server certificates)
2) We are not breaking any external APIs, this BiConsumer is a wrapper on whatever other components provide. Components provide a `GetCertificateRequest` which still uses a `Consumer` and that consumer receives a `CertificateUpdateEvent`  which we are still passing